kahuna/public/js/search/query.js

```diff
@@ -170,12 +175,27 @@ query.controller('SearchQueryCtrl', [
     // we can't user dynamic values in the ng:true-value see:
     // https://docs.angularjs.org/error/ngModel/constexpr
     mediaApi.getSession().then(session => {
+        const isNonFree = storage.getJs("isNonFree");
+        const isUploadedByMe = storage.getJs("isUploadedByMe");
+
         ctrl.user = session.user;
-        ctrl.filter.uploadedByMe = ctrl.uploadedBy === ctrl.user.email;
+        if (isUploadedByMe === undefined) {
+              ctrl.filter.uploadedByMe = ctrl.uploadedBy === ctrl.user.email;
+              storage.setJs("isUploadedByMe",ctrl.filter.uploadedByMe);
+        }
+        else {
+          ctrl.filter.uploadedByMe =  isUploadedByMe;
+        }
 
-        if (ctrl.filter.nonFree === undefined) {
+        if (isNonFree === undefined) {
```

Comment:
Am I right in thinking that this will break when users' permissions are revoked and `isNonFree` is true? If that's the case, perhaps we could return early if the permission is not present, setting `isNonFree` accordingly.